src/DynamoCore/Graph/Workspaces/SerializationConverters.cs

```diff
@@ -535,7 +535,8 @@ public override void WriteJson(JsonWriter writer, object value, JsonSerializer s
                 // a DSVarArgFunction or a CodeBlockNodeModel into a list.
                 var nodeGuids =
                     ws.Nodes.Where(
-                            n => n is DSFunction || n is DSVarArgFunction || n is CodeBlockNodeModel || n is Function)
+                            n => n is DSFunction || n is DSVarArgFunction || n is CodeBlockNodeModel || n is Function || 
+                            n.GetType().GetCustomAttributes(typeof(DynamoServices.RegisterForTraceAttribute),false).Length>0 )
```

Comment:
maybe .Any() would be more clear to - also, I wouldn't mind if this was a helper function. - we may already have it around in utils.

Reply:
is there any performance implication to adding this check to every node in the graph on save?

Reply:
Good points.  I will look for a preexisting utility.  As for perf I am not sure but I will check.  I suspect it can't be too much more as we are selecting and short-circuiting on every zero touch node.  There can't be too many nodes that don't fall under the first four conditions.  Let me look though if there is a faster way to test for the attribute.